src/SFA.DAS.EmployerAccounts/Queries/GetUnsignedEmployerAgreement/GetUnsignedEmployerAgreementValidator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerAccounts.Data;
+using SFA.DAS.Validation;
+
+namespace SFA.DAS.EmployerAccounts.Queries.GetUnsignedEmployerAgreement
+{
+    public class GetUnsignedEmployerAgreementValidator : IValidator<GetUnsignedEmployerAgreementRequest> 
+    {
+        private readonly IMembershipRepository _membershipRepository;
+
+        public GetUnsignedEmployerAgreementValidator(IMembershipRepository membershipRepository)
+        {
+            _membershipRepository = membershipRepository;
+        }
+
+        public ValidationResult Validate(GetUnsignedEmployerAgreementRequest item)
+        {
+            throw new NotImplementedException();
```

Comment:
As others are we purposely trying to block synchronous validation or can we just return the result of the async one ?